### PR TITLE
Adding Appliance Name Parameter to GetDiscoveredServer API

### PR DESCRIPTION
**Pull request by shivamverma-ms**
## Description

- Added Appliance Name as a parameter to the GetDiscoveredServer API to fetch the server discovered by that appliance, along with other parameters like displayname, projectName, etc.
- Added new Column in Input.csv as "AZMIGRATE_APPLIANCE_NAME".

Az version: 10.1.0

### CODE REVIEW

**Comment by shivamverma-ms** [comment]
@microsoft-github-policy-service agree company="Microsoft"

**Comment by shivamverma-ms** [comment]
> @shivamverma-ms please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

**Review by singhabh27 — COMMENTED** [review]
Remove extra log files.

File: azure-migrate/migrate-at-scale-vmware-agentles/Agentless VMware automation/AzMigrate_Shared.ps1 [hunk 1]
```diff
@@ -164,7 +165,7 @@ class AzMigrate_Shared
         $DiscoveredServer = $null
         $DiscoveredServerList = $null
         # Get Discovered machines in an Azure Migrate project
-        $DiscoveredServerList = Get-AzMigrateDiscoveredServer -ProjectName $AzMigrateProjectName `
+        $DiscoveredServerList = Get-AzMigrateDiscoveredServer -ApplianceName $AzMigrateApplianceName -ProjectName $AzMigrateProjectName `
```
Comment (singhabh27) [inline azure-migrate/migrate-at-scale-vmware-agentles/Agentless VMware automation/AzMigrate_Shared.ps1:168]: nit: Put -ProjectName $AzMigrateProjectName in next line

File: azure-migrate/migrate-at-scale-vmware-agentles/Agentless VMware automation/AzMigrate_StartTestMigration.ps1 [hunk 2]
```diff
@@ -102,6 +106,7 @@ Function ProcessItemImpl($processor, $csvItem, $reportItem) {
     #End Code for Target Subscription
 
     $TestMigrationJob = Start-AzMigrateTestMigration -InputObject $ReplicatingServermachine -TestNetworkID $Target_VNet.Id
+    process.Logger.LogTrace("TestMigrationJob: '$($TestMigrationJob)', and Target_VNET: '$($Target_VNet.Id)'")
```
Comment (singhabh27) [inline azure-migrate/migrate-at-scale-vmware-agentles/Agentless VMware automation/AzMigrate_StartTestMigration.ps1:109]: Remove..

File: azure-migrate/migrate-at-scale-vmware-agentles/Agentless VMware automation/Input.csv [hunk 3]
```diff
@@ -1 +1,3 @@
...
+﻿AZMIGRATEPROJECT_SUBSCRIPTION_ID,AZMIGRATEPROJECT_RESOURCE_GROUP_NAME,AZMIGRATEPROJECT_NAME,AZMIGRATE_APPLIANCE_NAME,SOURCE_MACHINE_NAME,AZMIGRATEASSESSMENT_NAME,AZMIGRATEGROUP_NAME,TARGET_SUBSCRIPTION_ID,TARGET_RESOURCE_GROUP_NAME,TARGET_VNET_NAME,TARGET_SUBNET_NAME,TARGET_MACHINE_NAME,TARGET_MACHINE_SIZE,LICENSE_TYPE,OS_DISK_ID,TARGET_DISKTYPE,AVAILABILITYZONE_NUMBER,AVAILABILITYSET_NAME,TURNOFF_SOURCESERVER,TESTMIGRATE_VNET_NAME,UPDATED_TARGET_RESOURCE_GROUP_NAME,UPDATED_TARGET_VNET_NAME,UPDATED_TARGET_MACHINE_NAME,UPDATED_TARGET_MACHINE_SIZE,UPDATED_AVAILABILITYZONE_NUMBER,UPDATED_AVAILABILITYSET_NAME,UPDATED_NIC1_ID,UPDATED_TARGET_NIC1_SELECTIONTYPE,UPDATED_TARGET_NIC1_SUBNET_NAME,UPDATED_TARGET_NIC1_IP,UPDATED_NIC2_ID,UPDATED_TARGET_NIC2_SELECTIONTYPE,UPDATED_TARGET_NIC2_SUBNET_NAME,UPDATED_TARGET_NIC2_IP,OK_TO_UPDATE,OK_TO_MIGRATE,OK_TO_USE_ASSESSMENT,OK_TO_TESTMIGRATE,OK_TO_RETRIEVE_REPLICATIONSTATUS,OK_TO_CLEANUP,OK_TO_TESTMIGRATE_CLEANUP
+xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx,srcrg-ecy,ECYPROJ,sv-app-2,testvm-cvm2,,srcrg-ecy,rg-name,sv-target-ecy-rg,vnet-name,default,testvm-cvm2,,NoLicenseType,,,,,Y,vnet-ecy,,,testvm-sv-2,,,,,,,,,,,,Y,Y,Y,Y,Y,Y,Y
```
Comment (singhabh27) [inline azure-migrate/migrate-at-scale-vmware-agentles/Agentless VMware automation/Input.csv:2]: Do we need to include sample inputs also?

Comment (shivamverma-ms) [reply azure-migrate/migrate-at-scale-vmware-agentles/Agentless VMware automation/Input.csv:2]: @prsadhu-ms-idc can you please check? If we can add the sample inputs.

**Review by mikefrobbins — APPROVED** [review]
LGTM

**Pull request merged**